contracts/0.8.25/vaults/VaultHub.sol

```diff
@@ -1376,64 +1312,63 @@ contract VaultHub is PausableUntilWithRoles {
         VaultObligations storage _obligations,
         uint256 _allowedUnsettled
     ) internal {
-        (
-            uint256 valueToRebalance,
-            uint256 sharesToRebalance,
-            uint256 valueToTransferToLido,
-            uint256 unsettledRedemptions,
-            uint256 unsettledLidoFees,
-            uint256 totalUnsettled
-        ) = _planSettlement(_vault, _record, _obligations);
-
-        // Enforce requirement for settlement completeness
-        if (totalUnsettled > _allowedUnsettled) {
-            revert VaultHasUnsettledObligations(_vault, totalUnsettled, _allowedUnsettled);
-        }
+        _verifySettlementPossibility(_vault, _record, _obligations, _allowedUnsettled);
 
-        // Skip if no changes to obligations
-        if (valueToTransferToLido == 0 && valueToRebalance == 0) {
-            return;
-        }
+        /// ORDER IS IMPORTANT HERE BECAUSE OF THE BALANCE REDUCTION IN THE REBALANCE FUNCTION
 
-        if (valueToRebalance > 0) {
-            _decreaseLiability(_vault, _record, sharesToRebalance);
-            _withdraw(_vault, _record, address(this), valueToRebalance);
-            _rebalanceExternalEtherToInternal(valueToRebalance);
+        /// 1. Rebalance
```

Comment:
Maybe it's worth to split this method in two:
- rebalance part can be integrated in forceRebalance
- payFees can be a separate function that do forceRebalance first